arquillian-chameleon-runner/runner/src/main/java/org/arquillian/container/chameleon/runner/ArquillianChameleonConfigurationGenerator.java

```diff
@@ -0,0 +1,104 @@
+
+            final Path tempDirectory = Files.createTempDirectory("chameleonContainer");
+            final Path arquillianPath = tempDirectory.resolve(ARQUILLIAN_PROPERTIES);
+
+            containerDefinitionAsProperties.store(new FileWriter(arquillianPath.toFile()), "");
+
+            return arquillianPath;
+
+        }
+
+        return null;
+    }
+
+    public Path generateNewArquillianXml(Class<?> testClass)
+        throws IOException, TransformerException {
+        final ChameleonTargetConfiguration chameleonTargetConfiguration = findChameleonTargetConfiguration(testClass);
+
+        if (chameleonTargetConfiguration != null) {
+
+            final Node containerDefinition = chameleonTargetConfiguration.getContainerDefinitionAsXml();
+            Document arquillianConfigurationDocument = DomManipulation.createDocumentFromTemplate();
+
+            return generate(arquillianConfigurationDocument, containerDefinition, ARQUILLIAN_XML);
+
+        }
+
+        return null;
+    }
+
+    public Path generateAppendedArquillianXml(Class<?> testClass, InputStream originalArquillianXml)
+        throws IOException, TransformerException {
+
+        final ChameleonTargetConfiguration chameleonTargetConfiguration = findChameleonTargetConfiguration(testClass);
+
+        if (chameleonTargetConfiguration != null) {
+
+            final Node containerDefinition = chameleonTargetConfiguration.getContainerDefinitionAsXml();
+            Document arquillianConfigurationDocument = DomManipulation.createDocumentFromInputStream(originalArquillianXml);
+
+            return generate(arquillianConfigurationDocument, containerDefinition, ARQUILLIAN_CHAMELEON_XML);
+
+        }
+
+        return null;
+
+    }
+
+    private Path generate(Document arquillianConfigurationDocument, Node containerDefinition, String filename)
+        throws IOException, TransformerException {
+        final Element arquillianElement = (Element) arquillianConfigurationDocument.getFirstChild();
+        final Node importNode = arquillianConfigurationDocument.importNode(containerDefinition, true);
+        arquillianElement.appendChild(importNode);
+
+        final Path tempDirectory = Files.createTempDirectory("chameleonContainer");
+        final Path arquillianPath = tempDirectory.resolve(filename);
+
+        DomManipulation.writeToPath(arquillianConfigurationDocument, arquillianPath);
+
+        return arquillianPath;
+    }
+
+    private ChameleonTargetConfiguration findChameleonTargetConfiguration(Class<?> testClass) {
+        final Annotation[] annotations = testClass.getAnnotations();
+
+        for (Annotation annotation : annotations) {
+            final ChameleonTargetConfiguration chameleonTargetConfiguration = AnnotationExtractor.extract(annotation);
+
+            if (chameleonTargetConfiguration != null) {
+                // We only support one container for now.
+                return chameleonTargetConfiguration;
```

Comment:
Shouldn't we warn if there is more then one defined? Or even fail instead of silently take the first one available?

Reply:
In fact it has no sense this comment since we are prior to Java8 you cannot annotate same class with same annotation multiple times.